ipwb/util.py

```diff
@@ -39,9 +42,26 @@
 dtPattern = re.compile(r"^(\d{4})(\d{2})?(\d{2})?(\d{2})?(\d{2})?(\d{2})?$")
 
 
-def isDaemonAlive(hostAndPort="{0}:{1}".format(IPFSAPI_HOST, IPFSAPI_PORT)):
+def getMultiAddress(hostAndPort=IPFSAPI_ADDRESS):
+    try:
+        (host, port) = hostAndPort.split(':')
+    except ValueError as err:
+        # Naive check to determine if multiaddr passed in
+        if hostAndPort[0] == '/' and hostAndPort.count('/'):
+            return hostAndPort
+        else:
+            print('Error reading IPFSAPI address')
+            sys.exit()
+    try:
+        socket.inet_aton(host)
```

Comment:
If we attempt to check the connection using the `ipfshttpclient` itself, then we will not need `socket` here and more importantly we will have no need to sore the `IPFSAPI_ADDRESS`. This whole `getMultiAddress` function can go away and be replaced with a simple constant holding the multi-address.

Reply:
The socket is usage is just a means for IP address validation. If it's not, the exception will be thrown.

The multiaddress will vary based on whether it's an IP address or host, per the `getMultiAddress` logic.

Reply:
> The socket is usage is just a means for IP address validation. If it's not, the exception will be thrown.

I understand the purpose and that is what I meant by `If we attempt to check the connection using the ipfshttpclient itself...` which can ensure that the can indeed connect to the provided multi-addr. If there is anything wrong, the client library will throw and exception. Using the client may come with many other corner cases covered. This custom validation is both limiting and incomplete. It enforces us to only allow a host and port or IPv4 and port style connection and nothing else, but we should leverage the flexibility we get from the multi-addr and offload any validation to the library itself.

Reply:
@ibnesayeed Instead of the logic in `getMultiAddress()` do you think we should instantiate an ipfshttpclient `client` instance, like we usually do, and if it can't connect (as indicated by a timeout, exception, error, etc), try a different multiaddress pattern?

The point of the function is to determine if the "host" is a hostname or IP address and generate the appropriately formatted mutliaddress.

If a client specifies an IP address or hostname on the CLI, the resulting multiaddress formatting will vary based on the "host".

Reply:
In the very first comment in the `__main__.py` file I suggested that we simply do not accept daemon address as IP addr/hostname + port number, Instead, we ask the user to supply a complete and valid multi-addr. This way, we can simply store the default multi-addr as a plain string and allow it to be overwritten completely by the CLI argument (if present). We have no business of storing it in pieces or validating in pieces. In order to validate the multi-addr, simply attempt to connect to it by creating a client using the function provided by the IPFS library. If there is any formatting or network issue, the library will through an exception that can be caught and reported. In case of success, we may choose to reuse this client or dispose it. This method does not require us to attempt for many different ways a multi-addr can be formatted, because we won't constructing it piece by piece.

Reply:
Ok, stand by. I'll implement that logic then give you a ping.

Reply:
@ibnesayeed Please review c9f48d3.